core/src/main/scala/org/apache/spark/scheduler/ExecutorDecommissionReason.scala

```diff
@@ -0,0 +1,90 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.scheduler
+
+private[spark] sealed trait ExecutorDecommissionReason {
+  val reason: String = "decommissioned"
```

Comment:
I don't think the `reason` field is really needed anywhere, besides it being used for `toString` ? Should we just require overriding `toString` by marking `toString` abstract ? I don't think that child classes need to override both `toString` and `reason` : I would prefer we just override methods instead of fields.

Reply:
We also indirectly use the reason for `logExecutorLoss`. And yeah we can just override toString indeed.